src/keywrap/jwe.rs

```diff
@@ -0,0 +1,230 @@
+use anyhow::{anyhow, Result};
+use std::collections::HashMap;
+
+use crate::config::DecryptConfig;
+use crate::config::EncryptConfig;
+use crate::keywrap::KeyWrapper;
+use josekit::jwk::{Jwk, KeyAlg, KeyFormat, KeyInfo};
+
+use josekit::jwe::{
+    deserialize_json, serialize_general_json, JweDecrypter, JweEncrypter, JweHeader, JweHeaderSet,
+    ECDH_ES_A256KW, RSA_OAEP,
+};
+
+/// A Jwe keywrapper
+#[derive(Debug)]
+pub struct JweKeyWrapper {}
+
+// Get the encrypter from public key
+fn get_encrypter(pubkey: &[u8]) -> Result<Box<dyn JweEncrypter>> {
+    let key_info = match KeyInfo::detect(&pubkey) {
+        Some(i) => i,
+        None => return Err(anyhow!("failed to detect public key info")),
+    };
+
+    if !key_info.is_public_key() {
+        return Err(anyhow!("expect public key, found private key"));
+    }
+
+    let alg = match key_info.alg().unwrap() {
```

Comment:
We want to propagate the error and avoid `unwrapping` through a `map_err` for example.